iOS-Weather-App/AppDelegate.swift

```diff
@@ -10,27 +10,45 @@ import UIKit
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
-
+    var window: UIWindow?
+    private var navigationController: UINavigationController?
+    private let weatherController = WeatherViewController()
+    private let searchViewModel = SearchViewModel()
+    private let currentDetailedWeatherModel = CurrentDetailedWeatherViewModel()
+    private let searchController = SearchViewController()
+    private let currentDetailsController = CurrentDetailedWeatherViewController()
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
```

Comment:
should put all logic there

Reply:
What logic? The ones I have in private extension as functions in app delegate?